packages/clerk-js/bundle-check.mjs

```diff
@@ -0,0 +1,238 @@
+ * creating an instance of `VirtualRouter`.
+ * @param {string} script The case-specific code to execute, typically to mount a component like `SignIn`.
+ */
+function template(script) {
+  return `
+<!DOCTYPE html>
+<html>
+    <body>
+        <div id="app"></div>
+        <script
+          type="text/javascript"
+          src="/clerk.browser.js"
+          data-clerk-publishable-key="${process.env.CLERK_PUBLISHABLE_KEY}"
+        ></script>
+        <script type="text/javascript">
+        class VirtualRouter {
+            name = 'VirtualRouter';
+            mode = 'virtual';
+
+            #url;
+            #listeners = new Set();
+
+            constructor(path) {
+                const origin = typeof window === 'undefined' ? 'https://clerk.dummy' : window.location.origin;
+
+                this.#url = new URL(path ?? '/', origin);
+            }
+
+            push(path) {
+                const newUrl = new URL(this.#url.toString());
+                newUrl.pathname = path;
+
+                this.#url = newUrl;
+                this.emit();
+            }
+
+            replace(path) {
+                this.push(path);
+            }
+
+            shallowPush(path) {
+                this.push(path);
+            }
+
+            pathname() {
+                return this.#url.pathname;
+            }
+
+            searchParams() {
+                return this.#url.searchParams;
+            }
+
+            subscribe(listener) {
+                this.#listeners.add(listener);
+
+                return () => this.#listeners.delete(listener);
+            }
+
+            emit() {
+                this.#listeners.forEach(listener => listener(this.#url));
+            }
+
+            getSnapshot() {
+                return this.#url;
+            }
+        }
+        window.VIRTUAL_ROUTER = new VirtualRouter('/');
+        </script>
+        <script type="text/javascript">${script}</script>
+    </body>
+</html>
+`;
+}
+
+function signIn() {
+  const script = `
+window.Clerk.load({ router: window.VIRTUAL_ROUTER }).then(() => {
+  window.Clerk.mountSignIn(document.getElementById("app"), {});
+});
+`;
+  return template(script);
+}
+
+function signUp() {
+  const script = `
+window.Clerk.load({ router: window.VIRTUAL_ROUTER }).then(() => {
+  window.Clerk.mountSignUp(document.getElementById("app"), {});
+});
+`;
+  return template(script);
+}
+
+/**
+ * Map of URL routes to functions that return the HTML content of the page. This script will generate a report for each
+ * of the routes defined here.
+ */
+const routes = {
+  '/sign-in': signIn(),
+  '/sign-up': signUp(),
+};
+
+const server = http
+  .createServer((req, res) => {
+    const onError = err => {
+      if (err) {
+        res.end();
+        console.error(err);
+      }
+    };
+
+    if (req.url && req.url in routes) {
+      res.writeHead(200, { 'content-type': 'text/html' });
+      res.end(routes[req.url]);
+    } else {
```

Comment:
Yeah this is actually because the TypeScript types are not great. `req.url` is _always_ defined for incoming messages received by the HTTP server. But I agree that there's no harm in still checking! I'll add that.